clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -256,12 +261,19 @@ private long listOffset(TopicPartition partition, long timestamp) {
      * Return the fetched records, empty the record buffer and update the consumed position.
      *
      * @return The fetched records per partition
+     * @throws OffsetOutOfRangeException If there is OffsetOutOfRange error in fetchResponse
      */
-    public Map<TopicPartition, List<ConsumerRecord<K, V>>> fetchedRecords() {
+    public Map<TopicPartition, List<ConsumerRecord<K, V>>> fetchedRecordsOrException() {
         if (this.subscriptions.partitionAssignmentNeeded()) {
             return Collections.emptyMap();
```

Comment:
I think we should not clear exception, because it should be handled in the same way we handle Fetcher.records -- we will still return already fetched records, instead of clearing it, after the rebalance. What do you think?